src/virtual-scroll.ts

```diff
@@ -278,28 +288,31 @@ export class VirtualScrollComponent implements OnInit, OnChanges, OnDestroy {
     end = Math.min(items.length, end);
     if (start !== this.previousStart || end !== this.previousEnd) {
 
-      // update the scroll list
-      this.viewPortItems = items.slice(start, end);
-      this.update.emit(this.viewPortItems);
+      this.zone.run(() => {
```

Comment:
`this.zone.run()` is nested inside an `if`, meaning that on the `else` case we won't run the zone. In such case, I think `scrollHeight` and `topPadding` won't be updated on the template because no CD will run. Using `zone.run()` outside the `if` would kill the perf so the best solution is probably to remove the bindings and use `Renderer2` to set the value programmatically.

Reply:
`this.scrollHeight` and `this.topPadding` only theoretically change if the start and end change, so I think it's okay. When the there is an actual change, the zone.run will cause CD which will detect the property changes.

Correct me if I am wrong here.

Reply:
They might also change if the items array changes, which would already cause a CD cycle.

Reply:
I'm don't fully understand your computations I just saw that you're setting  `this.scrollHeight` and `this.topPadding` before the `if` and their values seemed unrelated to `start`, `end`.
If their value are not important on the `else` case it's even better for perf to not sync their value with the DOM so all good then.